third_party/terraform/utils/transport.go

```diff
@@ -26,7 +26,7 @@ func isEmptyValue(v reflect.Value) bool {
 	case reflect.Array, reflect.Map, reflect.Slice, reflect.String:
 		return v.Len() == 0
 	case reflect.Bool:
-		return !v.Bool()
+		return false
```

Comment:
This doesn't make sense to me. It seems to be saying that booleans can _never_ be an empty value, which isn't true. False is the empty value for booleans, just like 0 is for ints and "" is for strings. I think what you actually want to do is set `send_empty_value: true` on the field in question.

Reply:
To cast a empty string to boolean, yes, it is false. But a false value of boolean type is concrete and clear, and it should be treated the same as true. But after comparing with values of other types, like treating Integer number 0 as `empty` from this function, I think my initial interpretation of the purpose of `isEmptyValue()` was different.